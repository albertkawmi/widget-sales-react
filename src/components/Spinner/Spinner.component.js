import React from 'react'
import logo from '../../images/spinner.png'
import './Spinner.css'

const Spinner = () => (
  <div className="spinner-outer">
    <img className="spinner" src={logo} alt="Loading..."/>
    <a className="spinner-link" href="/why-so-slow">Why is this taking so long?</a>
  </div>
)

export default Spinner
