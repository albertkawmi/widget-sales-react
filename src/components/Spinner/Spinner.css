.spinner-outer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 4rem;
}

.spinner {
  width: 40px;
  animation: spin 4s linear infinite;
}

.spinner-link {
  color: #2bafba;
  margin-top: 2rem;
}

@keyframes spin { 100% { transform: rotate(360deg); } }
