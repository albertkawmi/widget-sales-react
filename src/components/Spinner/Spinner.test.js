/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Spinner from './Spinner.component'

Enzyme.configure({ adapter: new Adapter() })

describe('<Spinner />', () => {
  it('renders a spinner', () => {
    const spinner = Enzyme.shallow(
      <Spinner />
    )
    expect(spinner).toMatchSnapshot()
  })
})
