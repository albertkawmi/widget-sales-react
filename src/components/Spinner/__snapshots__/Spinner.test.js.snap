// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders a spinner 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Loading..."
          className="spinner"
          src="spinner.png"
/>,
        <a
          className="spinner-link"
          href="/why-so-slow"
>
          Why is this taking so long?
</a>,
      ],
      "className": "spinner-outer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Loading...",
          "className": "spinner",
          "src": "spinner.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Why is this taking so long?",
          "className": "spinner-link",
          "href": "/why-so-slow",
        },
        "ref": null,
        "rendered": "Why is this taking so long?",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Loading..."
            className="spinner"
            src="spinner.png"
/>,
          <a
            className="spinner-link"
            href="/why-so-slow"
>
            Why is this taking so long?
</a>,
        ],
        "className": "spinner-outer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Loading...",
            "className": "spinner",
            "src": "spinner.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Why is this taking so long?",
            "className": "spinner-link",
            "href": "/why-so-slow",
          },
          "ref": null,
          "rendered": "Why is this taking so long?",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
